src/txmempool.cpp

```diff
@@ -1098,3 +1100,22 @@ void CTxMemPool::SetIsLoaded(bool loaded)
 }
 
 SaltedTxidHasher::SaltedTxidHasher() : k0(GetRand(std::numeric_limits<uint64_t>::max())), k1(GetRand(std::numeric_limits<uint64_t>::max())) {}
+
+std::string MemPoolRemovalReasonString(const MemPoolRemovalReason reason)
+{
+    switch (reason) {
+    case MemPoolRemovalReason::EXPIRY:
+        return "Expired from mempool";
```

Comment:
nit: I think these strings could just be `EXPIRY`, `SIZELIMIT`, etc. These strings are for logs only, so will only be read by engineers.

Trying to make these strings more generally accessible obscures some of the nuance. For example `MemPoolRemovalReason::REORG` means that a transaction was removed during a re-org because it spent a coinbase transaction that is no longer mature, or because it is no longer final, or because mempool policy changed since it first entered the mempool so it can no longer be added or because the re-org was too large and we dropped transactions to avoid memory exhaustion.